mne/preprocessing/faster.py

```diff
@@ -0,0 +1,368 @@
+
+
+def _hurst(x):
+    """Estimate Hurst exponent on a timeseries.
+
+    The estimation is based on the second order discrete derivative.
+
+    Parameters
+    ----------
+    x : 1D numpy array
+        The timeseries to estimate the Hurst exponent for.
+
+    Returns
+    -------
+    h : float
+        The estimation of the Hurst exponent for the given timeseries.
+    """
+    y = np.cumsum(np.diff(x, axis=1), axis=1)
+
+    b1 = [1, -2, 1]
+    b2 = [1,  0, -2, 0, 1]
+
+    # second order derivative
+    y1 = lfilter(b1, 1, y, axis=1)
+    y1 = y1[:, len(b1) - 1:-1]  # first values contain filter artifacts
+
+    # wider second order derivative
+    y2 = lfilter(b2, 1, y, axis=1)
+    y2 = y2[:, len(b2) - 1:-1]  # first values contain filter artifacts
+
+    s1 = np.mean(y1 ** 2, axis=1)
+    s2 = np.mean(y2 ** 2, axis=1)
+
+    return 0.5 * np.log2(s2 / s1)
+
+
+def _efficient_welch(data, sfreq):
+    """Calls scipy.signal.welch with parameters optimized for greatest speed
+    at the expense of precision. The window is set to ~10 seconds and windows
+    are non-overlapping.
+
+    Parameters
+    ----------
+    data : N-D numpy array
+        The timeseries to estimate signal power for. The last dimension
+        is presumed to be time.
+    sfreq : float
+        The sample rate of the timeseries.
+
+    Returns
+    -------
+    fs : 1D numpy array
+        The frequencies for which the power spectra was calculated.
+    ps : ND numpy array
+        The power spectra for each timeseries.
+    """
+    nperseg = min(data.shape[-1],
+                  2 ** int(np.log2(10 * sfreq) + 1))  # next power of 2
+
+    return welch(data, sfreq, nperseg=nperseg, noverlap=0, axis=-1)
+
+def _freqs_power(data, sfreq, freqs):
+    """Estimate signal power at specific frequencies.
+
+    Parameters
+    ----------
+    data : N-D numpy array
+        The timeseries to estimate signal power for. The last dimension
+        is presumed to be time.
+    sfreq : float
+        The sample rate of the timeseries.
+    freqs : list of float
+        The frequencies to estimate signal power for.
+
+    Returns
+    -------
+    ps : list of float
+        For each requested frequency, the estimated signal power.
+    """
+    fs, ps = _efficient_welch(data, sfreq)
+    return np.sum([ps[..., np.searchsorted(fs, f)] for f in freqs], axis=0)
+
+
+def _power_gradient(data, sfreq):
+    """Estimate the gradient of the power spectrum at upper frequencies.
+
+    Parameters
+    ----------
+    data : N-D numpy array
+        The timeseries to estimate signal power for. The last dimension
+        is presumed to be time.
+    sfreq : float
+        The sample rate of the timeseries.
+
+    Returns
+    -------
+    grad : N-D numpy array
+        The gradients of each timeseries.
+    """
+    fs, ps = _efficient_welch(data, sfreq)
+
+    # Limit power spectrum to upper frequencies
+    ps = ps[:, np.searchsorted(fs, 25):np.searchsorted(fs, 45)]
```

Comment:
because I'm lazy. This part needs more testing and looking in the original paper / EEGLAB code to find the exact procedure.